@types/index.d.ts

```diff
@@ -150,8 +150,11 @@ declare class Response extends Body {
 	readonly redirected: boolean;
 	readonly status: number;
 	readonly statusText: string;
+	readonly type: string;
```

Comment:
@tkrotoff `type` should be the `ResponseType` enum: https://fetch.spec.whatwg.org/#responsetype

Reply:
Should I
- use ResponseType from TypeScript lib.dom.d.ts
- create my own with just `type ResponseType = "default" | "error"`
- create my own with everything: `type ResponseType = "basic" | "cors" | "default" | "error" | "opaque" | "opaqueredirect"`

Reply:
Please create your own with everything from the spec.  While we only make use of `default` and `error`, they could be supported at some time in the future, and having them all maintains parity with the spec and other implementations.

Reply:
Done. However *could be supported at some time in the future* sounds YAGNI to me. Either use lib.dom.d.ts if you want to be future proof or provide your own types if you want to inform your users some stuffs are missing. I don't see the point of copy-pasting lib.dom.d.ts.

Reply:
@tkrotoff You make very good points.  Your thoughts about using lib.dom.d.ts are probably better suited for a larger refactor, and I wouldn't want to hold up this one.  I only have a cursory understanding of TypeScript, but I would recommend making a separate PR for that or opening an issue to ask about the appetite for such a refactor.  Also, I understand the YAGNI concern, and I would agree if this were a product. With it being a library, we want to ensure things can be augmented as needed.